applications/luci-app-statistics/root/etc/init.d/luci_statistics

```diff
@@ -21,16 +46,122 @@ start_service() {
 
 	### workaround broken permissions on /tmp
 	chmod 1777 /tmp
+
+	### restore if necessary
+	rrd_restore
+
+	### restart collectd if it was running before us
+	/etc/init.d/collectd status >/dev/null 2>&1 && /etc/init.d/collectd stop >/dev/null 2>&1
+	/etc/init.d/collectd start
```

Comment:
How about just having a "collectd restart" here?

currently this is actually not about "restart collectd if it was running before us", but it is "stop collectd if running + always start collectd".

If you want "restart collectd if it was running before us", it might be just "/etc/init.d/collectd status >/dev/null 2>&1 && /etc/init.d/collectd restart"

Reply:
Hmmm....
reading further, there actually was a "collectd" restart that you removed and replaced with a more polished logic. git diff hide that well at the first glance. 
Not sure if that is worthwhile. (I like KISS principle.)

Reply:
> Hmmm.... reading further, there actually was a "collectd" restart that you removed and replaced with a more polished logic. git diff hide that well at the first glance. Not sure if that is worthwhile. (I like KISS principle.)

This is what got lost when moving the docs to README and out of the source code comments.

The problem I found (really, this is unrelated to the backups/restores) is that if you install luci_statistics  with `opkg install luci_statistics`, it will load the dependencies first.  collectd gets installed and started with *its default configuration*.  Then luci_statistics gets installed and started, whereupon it creates a new config file.  But in this case, it never restarts collectd so the new config file is not yet used.

This problem does not occur if the system is built with both collectd and luci_statistics in the base image.  In that case, the first time it boots, luci_statistics goes first and sets up the config (with collectd not yet running), due to the /etc/rc.d/SXX ordering.

Reply:
> But in this case, it never restarts collectd so the new config file is not yet used.

Well, there was a "collectd restart" (that you removed) so it should restart once luci_statistics starts. Interesting if it does not happen.

> This problem does not occur if the system is built with both collectd and luci_statistics in the base image

Yeah, I have it always that way.

Reply:
> Well, there was a "collectd restart" (that you removed) so it should restart collectd once luci_statistics starts. Interesting if it does not happen.

That previous restart of collectd is only inside the code to restart luci_statistics, not in the code to *start* luci_statistics

Reply:
> currently this is actually not about "restart collectd if it was running before us", but it is "stop collectd if running + always start collectd".
> 
> If you want "restart collectd if it was running before us", it might be just "/etc/init.d/collectd status >/dev/null 2>&1 && /etc/init.d/collectd restart"

I think we want "stop if running and then start".  Otherwise, if collectd is not actually running, then starting just luci_statistics won't really start gathering statistics.  In the current code (before this PR), the sysadmin needs to know that if both collectd & luci_statistcs are stopped, then they have to start luci_statistics and then collectd in order to get statistics gathering started.

So I can either update the comment to explain that we want "stop collectd if necessary and always start it", or leave the comment and change the code to "restart collectd if it was running".  I of course prefer the former.